packages/dina-ui/components/collection/material-sample/useMaterialSample.tsx

```diff
@@ -697,7 +690,10 @@ export function useMaterialSampleSave({
               pick(it, "id", "type")
             )
           }
-        })
+        }),
+        preparedBy: {
+          data: preparedBy.map((it) => pick(it, "id", "type"))
+        }
```

Comment:
This is exactly the right idea, setting it here then deleting the attribute version.

But we should still have a check to see if it's been added to the msDiffWithOrganisms because we don't want to include it in the request again if it's already been set. We only want to have the things that have changed in the request.

That way if the user does an edit, and does not touch the preparedBy it will not be added to the request and will not be altered. Only fields that are touched will be included in the PATCH request.

Something like this:

```typescript
     ...(msDiffWithOrganisms.preparedBy && {
        preparedBy: {
          data: msDiffWithOrganisms.preparedBy.map((it) => pick(it, "id", "type"))
        }
    })
```
Then below it can be set undefined with the others:
```typescript
      // Set the attributes to undefined after they've been moved to "relationships":
      attachment: undefined,
      projects: undefined,
      organism: undefined,
      assemblages: undefined,
      preparedBy: undefined
```